edx/edx-enterprise
# [ENT-640] Ensure all Enterprise/Consent API endpoints apply appropriate permissioning/filtering scheme

**Description:** This PR adds proper permission/filter schemes to our API endpoints.

**JIRA:** [ENT-640](https://openedx.atlassian.net/browse/ENT-640)

**Merge deadline:** ASAP

**Testing instructions:**

Please see the thread on [the ticket](https://openedx.atlassian.net/browse/ENT-640) to understand how our endpoints should actually be in fact behaving. Then, go through each of the APIs as a staff & non-staff user, and at each step ensuring you get the expected results.

Note that we will **skip** the consent API for testing here -- no changes were made to it that should affect its current functionality in any way, except making sure it works in cases where it may have normally broke (i.e. getting back-up values).

As a **staff user**, go through the following endpoints and ensure you get back all objects for that endpoint:

1. `/enterprise/api/v1/enterprise-catalogs/`
1. `/enterprise/api/v1/enterprise-course-enrollment/`
1. `/enterprise/api/v1/enterprise-customer/`
1. `/enterprise/api/v1/enterprise-customer/{uuid}/courses/`
1. `/enterprise/api/v1/enterprise-customer-branding/`
1. `/enterprise/api/v1/enterprise-customer-entitlement/`
1. `/enterprise/api/v1/enterprise-learner/`
1. `/enterprise/api/v1/catalogs/`

Now make a fresh **non-staff** user (confirm activation email) who isn't connected to any Enterprise, and get any Enterprise UUID for testing. Go through the following endpoints:

1. `/enterprise/api/v1/enterprise-catalogs/` -- ensure that you see **no catalogs**.
1. `/enterprise/api/v1/enterprise-course-enrollment/` -- ensure that you see **no enrollments**.
1. `/enterprise/api/v1/enterprise-customer/` -- ensure that you see **no customers**.
1. `/enterprise/api/v1/enterprise-customer/{uuid}/courses/` -- ensure that you see **no customer courses**.
1. `/enterprise/api/v1/enterprise-customer-branding/` -- ensure that you see **no brandings**.
1. `/enterprise/api/v1/enterprise-customer-entitlement/` -- ensure that you see **no entitlements**.
1. `/enterprise/api/v1/enterprise-learner/` -- ensure that you see **no learners**.
1. `/enterprise/api/v1/catalogs/` -- ensure that you see **no catalogs**.

Now make your **non-staff** user become connected to some Enterprise. Get the UUID for testing. Go through the following endpoints:

1. `/enterprise/api/v1/enterprise-catalogs/` -- ensure that you see only the **catalogs of the customer you're associated with**.
1. `/enterprise/api/v1/enterprise-course-enrollment/` -- ensure that you see **no enrollments** (because you've not enrolled in anything yet).
1. `/enterprise/api/v1/enterprise-customer/` -- ensure that you only see the **customer you're associated with**.
1. `/enterprise/api/v1/enterprise-customer/{uuid}/courses/` -- ensure that you see **no courses** (unable to fetch them because your user isn't given permissions for discovery).
1. `/enterprise/api/v1/enterprise-customer-branding/` -- ensure that you only see the **branding of the customer you're associated with**.
1. `/enterprise/api/v1/enterprise-customer-entitlement/` -- ensure that you only see the **entitlements of the customer you're associated with**.
1. `/enterprise/api/v1/enterprise-learner/` -- ensure that you only see **yourself**.
1. `/enterprise/api/v1/catalogs/` -- ensure that you see **no catalogs** (unable to fetch them because your user isn't given permissions for discovery).

Now go to the course enrollment landing page for any course for your Enterprise, and enroll in it with your **non-staff** user.

1. `/enterprise/api/v1/enterprise-course-enrollment/` -- ensure that you see **that enrollment** only.

**Merge checklist:**

  **DO NOT** just add dependencies to `requirements/*.txt` files.

**Post merge:**
